### ISSUE TITLE
Linux install script not working with Python 3.10

### ISSUE THREAD

**Issue by pallxk** [Azure/azure-cli#20713]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**

The install script `https://aka.ms/InstallAzureCli` does not work with python >= 3.10

**To Reproduce**

```
docker run --rm -it python:3.10 bash

curl -L https://aka.ms/InstallAzureCli | bash
```

**Expected behavior**

It installs.

**Environment summary**

```
root@d4a94dc7ef92:/# python --version
Python 3.10.1
root@d4a94dc7ef92:/# pip --version
pip 21.2.4 from /usr/local/lib/python3.10/site-packages/pip (python 3.10)
```

**Additional context**

https://github.com/Azure/azure-cli/blob/azure-cli-2.31.0/scripts/curl_install_pypi/install.py#L44-L47

https://github.com/pypa/virtualenv/pull/2109

https://virtualenv.pypa.io/en/legacy/changes.html#v16-7-11-2021-07-20

Virutalenv is fixed to 16.7.7, it seems we need at least 16.7.11

**Comment by jiasli** [comment]
Thanks for the contribution. I will take a look.

**Comment by pallxk** [comment]
Fixed in 2.32.0
